java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -287,6 +289,15 @@ private void visit(Tree tree, Tree terminator) {
           programState = ProgramState.put(programState, variableTree.symbol(), unstack.b.get(0));
         }
         break;
+      case TYPE_CAST:
+        TypeCastTree typeCast = (TypeCastTree) tree;
+        Type type = typeCast.type().symbolType();
+        if (type.isPrimitive()) {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 1);
+          programState = unstack.a;
+          programState = ProgramState.stackValue(programState, constraintManager.createSymbolicValue(typeCast.expression()));
```

Comment:
I tend to think that we shoud create the new symbolic value based on the type cast tree and not the expression.